### ISSUE TITLE
find-symbol leads to OutOfMemoryError

### ISSUE THREAD

**Issue by mbuczko** [clojure-emacs/refactor-nrepl#278]
##  Expected behavior
`find-symbol` should return list of symbol occurences

## Actual behavior
In some cases `find-symbol` ends up enormous memory consumption resulting in OutOfMemoryError. Partial results are returned.

This issue may be a duplicate of https://github.com/clojure-emacs/refactor-nrepl/issues/245

Following is the stack trace:

```clojure
#error {
 :cause Java heap space
 :via
 [{:type java.lang.OutOfMemoryError
   :message Java heap space
   :at [java.util.Arrays copyOf Arrays.java 3746]}]
 :trace
 [[java.util.Arrays copyOf Arrays.java 3746]
  [java.lang.AbstractStringBuilder ensureCapacityInternal AbstractStringBuilder.java 227]
  [java.lang.AbstractStringBuilder append AbstractStringBuilder.java 593]
  [java.lang.StringBuffer append StringBuffer.java 306]
  [java.io.StringWriter write StringWriter.java 106]
  [clojure.core$fn__7280 invokeStatic core_print.clj 124]
  [clojure.core$fn__7280 invoke core_print.clj 123]
  [clojure.lang.MultiFn invoke MultiFn.java 234]
  [clojure.core$pr_on invokeStatic core.clj 3674]
  [clojure.core$pr_on invoke core.clj 3668]
  [clojure.core$print_prefix_map$fn__7332 invoke core_print.clj 233]
  [clojure.core$print_sequential invokeStatic core_print.clj 66]
  [clojure.core$print_prefix_map invokeStatic core_print.clj 229]
  [clojure.core$print_map invokeStatic core_print.clj 238]
  [clojure.core$fn__7359 invokeStatic core_print.clj 266]
  [clojure.core$fn__7359 invoke core_print.clj 263]
  [clojure.lang.MultiFn invoke MultiFn.java 234]
  [clojure.core$pr_on invokeStatic core.clj 3674]
  [clojure.core$pr_on invoke core.clj 3668]
  [clojure.core$print_prefix_map$fn__7332 invoke core_print.clj 233]
  [clojure.core$print_sequential invokeStatic core_print.clj 66]
  [clojure.core$print_prefix_map invokeStatic core_print.clj 229]
  [clojure.core$print_map invokeStatic core_print.clj 238]
  [clojure.core$fn__7359 invokeStatic core_print.clj 266]
  [clojure.core$fn__7359 invoke core_print.clj 263]
  [clojure.lang.MultiFn invoke MultiFn.java 234]
  [clojure.core$pr_on invokeStatic core.clj 3674]
  [clojure.core$pr_on invoke core.clj 3668]
  [clojure.core$print_prefix_map$fn__7332 invoke core_print.clj 233]
  [clojure.core$print_sequential invokeStatic core_print.clj 66]
  [clojure.core$print_prefix_map invokeStatic core_print.clj 229]
  [clojure.core$print_map invokeStatic core_print.clj 238]]}
```

This is memory usage right before calling a `find-symbol`:

<img width="797" alt="main_-_YourKit_Java_Profiler_2018_04-b88_-_64-bit" src="https://user-images.githubusercontent.com/586537/70810716-8c212780-1dc4-11ea-8343-5ee6468f0d4f.png">

and here is how memory consumption looks like during- and after `find-symbol`:

<img width="792" alt="main_-_YourKit_Java_Profiler_2018_04-b88_-_64-bit" src="https://user-images.githubusercontent.com/586537/70810792-afe46d80-1dc4-11ea-825c-c7a714779117.png">

**Comment by mbuczko** [comment]
After a couple of days with profiler I finally found a culprit:

https://github.com/clojure-emacs/refactor-nrepl/blob/75f5255f0e4e701f50d30cc58216e03c2d558905/src/refactor_nrepl/find/find_symbol.clj#L27

It looks quite innocent, but the problem is what `full-class` may really be. It's usually a string (or something easily casted to string) although in certain cases it might be also a huge map (I guest AST node) which`str/join` is not able easily serialize to string. Looks like failing "stringification" leads to large memory leaks here.

I prepared already a fix already which eliminates a leak and keeps memory consuption sane:

<img width="794" alt="main_-_YourKit_Java_Profiler_2018_04-b88_-_64-bit" src="https://user-images.githubusercontent.com/586537/70811574-26359f80-1dc6-11ea-836a-924b53c1d94f.png">
